spec/std/spec/filters_spec.cr

```diff
@@ -0,0 +1,141 @@
+require "spec"
+
+module Spec::Item
+  setter focus : Bool
+end
+
+private class FakeRootContext < Spec::RootContext
+  def description
+    "root"
+  end
+end
+
+private def all_descriptions(item) : Array(String)
+  child_descriptions = item.responds_to?(:children) ? item.children.flat_map { |c| all_descriptions(c) } : [] of String
+  child_descriptions.unshift(item.description)
+end
+
+private def build_spec(filename, root = nil, count = 2)
```

Comment:
I was trying to find a way to parse a heredoc of an example file, but even if I did, I wouldn't know how to tell it to use the alternate `FakeRootContext` instead if `Spec.root_context`.  This approach here is similar to what's done in the other specs, so I thought it was good enough.

---

This method produces the equivalent of a spec file named `f.cr` with:

```crystal
context "context_f_1" do
  it "example_f_1_1" do
    # noop
  end

  it "example_f_1_2" do
    # noop
  end
end

context "context_f_2" do
  it "example_f_2_1" do
    # noop
  end

  it "example_f_2_2" do
    # noop
  end
end
```